### ISSUE TITLE
[apex] Excessive Method Length

### ISSUE THREAD

**Issue by mitchspano** [pmd/pmd#5429]
**Proposed Rule Name:** [apex] Excessive Method Length

**Proposed Category:** Design

**Description:**  This rule promotes writing concise and focused methods by enforcing a maximum line limit.  Long methods can be difficult to understand, test, and maintain.  Keeping methods short improves code readability and encourages breaking down complex logic into smaller, more manageable units.

**Code Sample:**

```java
// Violates the rule (assuming maxLines = 45): Method exceeds 45 lines
public void myLongMethod() {
    // Line 1
    // Line 2
    // ...
    // Line 45
    // Line 46
}

// Compliant code: Method is within the line limit
public void myConciseMethod() {
    // Line 1
    // Line 2
}
```

**Possible Properties:**

* **maxLines**: (Integer): The maximum number of lines allowed in a method. Default value is 45. This allows for customization based on project-specific coding standards.
* **maxLinesInTest**: (Integer): The maximum number of lines allowed in a method within a test class. Default value is 100.
* **IgnoreTestClasses**: (Boolean): A flag to indicate whether the rule should be ignored in test classes. Test methods sometimes require more setup and assertions, which can lead to longer method bodies.

**Comment by sirephil** [comment]
Instead of using the term "lines" can we use "statements"?

**Comment by Damecek** [comment]
We already have [ExcessiveClassLength](https://pmd.github.io/pmd/pmd_rules_apex_design.html#excessiveclasslength). Perhaps we could merge  it with a similar concept into a single rule, `ExcessiveLength`, which would include two properties `classReportLevel` and `methodReportLevel `. This would align with the approach taken for [CyclomaticComplexity](https://pmd.github.io/pmd/pmd_rules_apex_design.html#cyclomaticcomplexity) and [CognitiveComplexity](https://pmd.github.io/pmd/pmd_rules_apex_design.html#cognitivecomplexity), as all three rules aim to address a similar issue: reducing excessive code length.
